pipelines/broad/arrays/imputation/Imputation.wdl

```diff
@@ -0,0 +1,357 @@
+            basename = "chrom" + referencePanelContig.contig + "_chunk_" + i +"_imputed",
+            bcftools_docker = bcftools_docker_tag
+        }
+      }
+    }
+    Array[File] aggregatedImputationMetrics = select_all(AggregateImputationQCMetrics.aggregated_metrics)
+    Array[File] chromosome_vcfs = select_all(SortIds.output_vcf)
+    Array[File] chromosome_vcf_indices = select_all(SortIds.output_vcf_index)
+  }
+
+  Array[File] phased_vcfs = flatten(chromosome_vcfs)
+  Array[File] phased_vcf_indices = flatten(chromosome_vcf_indices)
+
+  call tasks.GatherVcfs {
+    input:
+      input_vcfs = phased_vcfs,
+      input_vcf_indices = phased_vcf_indices,
+      output_vcf_basename = output_callset_name,
+      gatk_docker = gatk_docker_tag
+  }
+
+  call tasks.ExtractIDs {
+    input:
+      vcf = GatherVcfs.output_vcf,
+      output_basename = "imputed_sites",
+      bcftools_docker = bcftools_docker_tag
+  }
+
+  call tasks.FindSitesFileTwoOnly {
+    input:
+      file1 = ExtractIDs.ids,
+      file2 = ExtractIdsVcfToImpute.ids,
+      ubuntu_docker = ubuntu_docker_tag
+  }
+
+  call tasks.SelectVariantsByIds {
+    input:
+      vcf = SortIdsVcfToImpute.output_vcf,
+      ids = FindSitesFileTwoOnly.missing_sites,
+      basename = "imputed_sites_to_recover",
+      gatk_docker = gatk_docker_tag
+  }
+
+  call tasks.RemoveAnnotations {
+    input:
+      vcf = SelectVariantsByIds.output_vcf,
+      basename = "imputed_sites_to_recover_annotations_removed",
+      bcftools_docker = bcftools_docker_tag
+  }
+
+  call tasks.InterleaveVariants {
+    input:
+      vcfs = [RemoveAnnotations.output_vcf, GatherVcfs.output_vcf],
+      basename = output_callset_name,
+      gatk_docker = gatk_docker_tag
+  }
+
+  call tasks.MergeImputationQCMetrics {
+    input:
+      metrics = flatten(aggregatedImputationMetrics),
+      basename = output_callset_name,
+      rtidyverse_docker = rtidyverse_docker_tag
+  }
+
+  if (MergeImputationQCMetrics.frac_well_imputed < frac_well_imputed_threshold) {
+    call tasks.FailQCThreshold as FailQCWellImputedFrac {
+      input:
+        msg = "Well imputed fraction was " + MergeImputationQCMetrics.frac_well_imputed + ", QC failure threshold was set at " + frac_well_imputed_threshold
+    }
+  }
+
+  call tasks.StoreChunksInfo {
+    input:
+      chroms = flatten(chunk_contig),
+      starts = flatten(start),
+      ends = flatten(end),
+      vars_in_array = flatten(CountChunks.var_in_original),
+      vars_in_panel = flatten(CountChunks.var_in_reference),
+      valids = flatten(CheckChunks.valid),
+      basename = output_callset_name,
+      rtidyverse_docker = rtidyverse_docker_tag
+  }
+
+  if (StoreChunksInfo.n_failed_chunks >= chunks_fail_threshold) {
+    call tasks.FailQCThreshold as FailQCNChunks {
+      input:
+        msg = StoreChunksInfo.n_failed_chunks + " chunks failed imputation, QC threshold was set to " + chunks_fail_threshold
+    }
+  }
+
+  call tasks.CrosscheckFingerprints {
```

Comment:
Why CrosscheckFingerprints?  Is this error checking?

Reply:
Yeah just a sanity check, but honestly it's not really necessary.  I could certainly be convinced it should be removed.  But it's also a very cheap step, so I'm pretty ambivalent about it one way or the other.